### PR TITLE
use SLY to tokenize input command stream

### PR DESCRIPTION
**Pull request by crhan**
use compiler technique to parse input stream to token stream

### CODE REVIEW

**Comment by crhan** [comment]
验证代码看这里... https://gist.github.com/crhan/b208009a85176f27d3fe027a2bc1b5b2

File: tests/unittests/test_utils.py [hunk 1]
```diff
@@ -52,7 +52,7 @@ def test_timer():
         (r""" "hello\"world" """, ['hello"world']),
         (r"''", [""]),  # set foo "" is a legal command
         (r'""', [""]),  # set foo "" is a legal command
-        (r"\\", ["\\\\"]),  # blackslash are legal
```
Comment (laixintao) [inline tests/unittests/test_utils.py]: Why need to delete this test?

Comment (crhan) [reply tests/unittests/test_utils.py]: case on line 46 is a equivalent case

**Comment by laixintao** [comment]
Hi @crhan , I am going to close this, since it introduced a new dependency which I don't think it's necessary. But it's a nice work! You showed another undeveloped possible solution. Feel free to comment if you have further ideas.

**Pull request closed without merging**